src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/NativeMethods.cs

```diff
@@ -533,29 +511,29 @@ PHANDLE LsaHandle
               PULONG ReturnBufferLength,
               PNTSTATUS ProtocolStatus
             );*/
-        [DllImport(global::Interop.Libraries.Secur32)]
-        internal static extern int LsaCallAuthenticationPackage(
-            [In] LsaLogonProcessSafeHandle lsaHandle,
-            [In] int authenticationPackage,
-            [In] NegotiateCallerNameRequest protocolSubmitBuffer,
-            [In] int submitBufferLength,
-            [Out] out IntPtr protocolReturnBuffer,
-            [Out] out int returnBufferLength,
-            [Out] out int protocolStatus);
+        [GeneratedDllImport(global::Interop.Libraries.Secur32)]
+        internal static partial uint LsaCallAuthenticationPackage(
+            LsaLogonProcessSafeHandle lsaHandle,
+            int authenticationPackage,
+            in NegotiateCallerNameRequest protocolSubmitBuffer,
```

Comment:
How are we choosing when to use 'in' and when to use 'ref'?  There was another recent PR where this was discussed and IIRC that one stayed `ref` even though it was previously similarly '[In]'.

Reply:
It has been somewhat opportunistic on our end as a `could be nice to do` thing where we change it if we notice it would make sense to be `in` instead of `ref` (in a recent PR, when @jkoritzinsky pointed it out, I did switch the `ref` to `in`).